libraries/protocol/smt_operations.cpp

```diff
@@ -198,23 +198,38 @@ void smt_setup_emissions_operation::validate()const
    
    FC_ASSERT( schedule_time > STEEM_GENESIS_TIME );
    FC_ASSERT( emissions_unit.token_unit.empty() == false );
+
+   for ( const auto& e : emissions_unit.token_unit )
+      validate_account_name( e.first );
+
+   FC_ASSERT( interval_seconds >= SMT_EMISSION_MIN_INTERVAL_SECONDS,
+      "Interval seconds must be greater than or equal to ${seconds}", ("seconds", SMT_EMISSION_MIN_INTERVAL_SECONDS) );
+
+   FC_ASSERT( interval_count > 0, "Interval count must be greater than 0" );
    
-   //interval_seconds <- any value of unsigned int is OK
-   //interval_count <- any value of unsigned int is OK
-   
-   FC_ASSERT( lep_time <= rep_time );
-   FC_ASSERT( schedule_time <= lep_time || lep_time == rep_time );
-   // ^ lep_time is either later or non-important
+   FC_ASSERT( lep_time <= rep_time, "Left endpoint time must be less than or equal to right endpoint time" );
+
+   // If time modulation is enabled
+   if ( lep_time != rep_time )
+   {
+      FC_ASSERT( lep_time >= schedule_time, "Left endpoint time cannot be before the schedule time" );
+
+      // If we don't emit indefinitely
+      if ( interval_count != SMT_EMIT_INDEFINITELY )
+      {
+         FC_ASSERT( rep_time <= schedule_time + fc::seconds( uint64_t( interval_seconds ) * uint64_t( interval_count ) ),
```

Comment:
You overflow checks don't solve the real problem that `fc::time_point_sec` is a 32 bit timestamp.

You need to ensure that `uint64_t( interval_seconds ) * uint64_t( interval_count ) + schedule_time.sec_since_epoch() <= std::numeric_limits< int32_t >::max` or else the time might wrap. We are pretty lose with using fc::time_point_sec vs fc::time_point_sec::sec_since_epoch and I am not convinced there isn't behavior that would fail in one way or work in the other. Ensuring the end time does not wrap is safest way to know that regardless of how the end time is calculated that it does not overflow.